geth/rpc/call_raw_test.go

```diff
@@ -67,26 +68,57 @@ func TestMethodAndParamsFromBody(t *testing.T) {
 			},
 			"subtract",
 			json.RawMessage(`42`),
+			false,
 		},
 		{
 			"params_empty_array",
 			`{"jsonrpc": "2.0", "method": "test", "params": []}`,
 			[]interface{}{},
 			"test",
 			json.RawMessage(nil),
+			false,
 		},
 		{
 			"params_none",
 			`{"jsonrpc": "2.0", "method": "test"}`,
 			[]interface{}{},
 			"test",
 			json.RawMessage(nil),
+			false,
+		},
+		{
+			"getFilterMessage",
+			`{"jsonrpc":"2.0","id":44,"method":"shh_getFilterMessages","params":["3de6a8867aeb75be74d68478b853b4b0e063704d30f8231c45d0fcbd97af207e"]}`,
+			[]interface{}{string("3de6a8867aeb75be74d68478b853b4b0e063704d30f8231c45d0fcbd97af207e")},
+			"shh_getFilterMessages",
+			json.RawMessage(`44`),
+			false,
+		},
+		{
+			"getFilterMessage_array",
+			`[{"jsonrpc":"2.0","id":44,"method":"shh_getFilterMessages","params":["3de6a8867aeb75be74d68478b853b4b0e063704d30f8231c45d0fcbd97af207e"]}]`,
+			[]interface{}{string("3de6a8867aeb75be74d68478b853b4b0e063704d30f8231c45d0fcbd97af207e")},
+			"shh_getFilterMessages",
+			json.RawMessage(`44`),
+			false,
+		},
+		{
+			"empty_array",
+			`[]`,
+			[]interface{}{},
+			"",
+			json.RawMessage(nil),
+			true,
 		},
 	}
 
 	for _, test := range cases {
 		t.Run(test.name, func(t *testing.T) {
 			method, params, id, err := methodAndParamsFromBody(test.body)
+			if test.shouldFail {
+				require.Error(t, err)
+				return
```

Comment:
require package will terminate automatically, no need to return.

Reply:
You liked but not fixed :) Anyway, merging.

Reply:
@tiabc I did :) See commit below.

Reply:
Missed that but saw the rest :(